src/overlays/actors/ovl_En_Bba_01/z_en_bba_01.c

```diff
@@ -33,84 +42,296 @@ const ActorInit En_Bba_01_InitVars = {
+                      &this->enHy.headRot, &this->enHy.torsoRot, D_809CCCD8);
+    } else {
+        Math_SmoothStepToS(&this->enHy.focusTarget.x, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->enHy.focusTarget.y, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->enHy.headRot.x, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->enHy.headRot.y, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->enHy.torsoRot.x, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->enHy.torsoRot.y, 0, 4, 0x3E8, 1);
+    }
+    SubS_FillLimbRotTables(globalCtx, this->enHy.limbRotTableY, this->enHy.limbRotTableZ,
+                           ARRAY_COUNT(this->enHy.limbRotTableY));
+    EnHy_UpdateCollider(&this->enHy, globalCtx);
+}
+
+s32 EnBba01_TestIsTalking(EnBba01* this, GlobalContext* globalCtx) {
+    s32 isTalking = false;
+
+    if (Actor_ProcessTalkRequest(&this->enHy.actor, &globalCtx->state)) {
+        isTalking = true;
+        this->enHy.textId = 0x10B9;
+        this->enHy.tmpFocusTarget = this->enHy.focusTarget;
+        this->enHy.tmpHeadRot = this->enHy.headRot;
+        this->enHy.tmpTorsoRot = this->enHy.torsoRot;
+        this->enHy.tmpActionFunc = this->enHy.actionFunc;
+        this->enHy.actionFunc = EnBba01_Talk;
+    }
+    return isTalking;
+}
+
+s32 func_809CC270(EnBba01* this, GlobalContext* globalCtx) {
+    s16 x;
+    s16 y;
+
+    Actor_GetScreenPos(globalCtx, &this->enHy.actor, &x, &y);
+    //! @bug: Both x and y conditionals are always true, || should be an &&
+    if (!this->enHy.waitingOnInit && ((x >= 0) || (x < SCREEN_WIDTH)) && ((y >= 0) || (y < SCREEN_HEIGHT))) {
+        func_800B85E0(&this->enHy.actor, globalCtx, 30.0f, EXCH_ITEM_2E);
+    }
+    return true;
+}
+
+void EnBba01_FinishInit(EnHy* this, GlobalContext* globalCtx) {
+    //! @bug: gBbaSkel does not match EnHy's skeleton assumptions.
+    //! Since gBbaSkel has more limbs than expected, joint and morph tables will overflow
+    if (EnHy_Init(this, globalCtx, &gBbaSkel, ENHY_ANIMATION_BBA_6)) {
+        this->actor.flags |= ACTOR_FLAG_1;
+        this->actor.draw = EnBba01_Draw;
+        this->waitingOnInit = false;
+        if (ENBBA01_GET_PATH(&this->actor) == ENBBA01_NO_PATH) {
+            this->actionFunc = EnBba01_FaceFoward;
+        } else {
+            this->actionFunc = EnBba01_Walk;
+        }
+    }
+}
+
+void EnBba01_Walk(EnHy* this, GlobalContext* globalCtx) {
+    if (EnHy_MoveForwards(this, 1.0f)) {
+        this->curPoint = 0;
+    }
+}
+
+void EnBba01_FaceFoward(EnHy* this, GlobalContext* globalCtx) {
+    this->actor.shape.rot = this->actor.world.rot;
+}
+
+void EnBba01_Talk(EnHy* this, GlobalContext* globalCtx) {
+    s16 yaw;
+    u8 talkState;
+
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 4, 0xFA0, 1);
+    talkState = Message_GetState(&globalCtx->msgCtx);
+    this->inMsgState3 = (talkState == 3) ? true : false;
```

Comment:
I'm guessing because ternary follows logic, this cannot be replaced with just the expression? (IDO isn't smart enough to know they are the same)
```
this->inMsgState3 = (talkState == 3);
```

Reply:
Yeah, different codegen. If you take look at my SubS part 5 PR, there are two more functions that require this same structure, so it seems at least one dev used this style.